### PR TITLE
Add cards task 47

### PR DESCRIPTION
**Pull request by osbbarr1**
The task is carried out again with the new architecture of the Carpa project

Oscar Baquero :)

### CODE REVIEW

**Comment by diferpaho** [comment]
Los cambios son correctos pero The changes are correct but bunch, bunchbatch, bunchcategory and sensor would go in the harvests application and you are adding it in sources, please add them in the harvests application, the origin-> Parcel changes look good

**Comment by osbbarr1** [comment]
Goodnight
this activity was carried out in the update-view-name-origin pullreques, 

I don't understand why it's time to do it again.?

**Comment by diferpaho** [comment]
> Goodnight
> this activity was carried out in the update-view-name-origin pullreques,
> 
> I don't understand why it's time to do it again.?

Each application has its models, buch and the others that I mentioned in the previous comment are from the harvest application, you must take into account in which application each model is, I leave you a number of the locations:

harvest:
-Bunch
-CategoryBunch
-Harvester
-BunchBatch

sources:
-Vehicle
-Driver
-Parcel
-ParcelOwner

you need to verify that the changes are made in the correct place, for example:
-`test_parcel_GET` you are creating it in `main/tests / test_views.py` and it should be in `sources / tests / test_views.py`
-the `bundle.html` is creating it in `sources / templates / bunch.html` and it should be in `harvest / templates / bunch.html`

File: main/tests/test_views.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 class TestViews(TestCase):
 
-
-
-
+    def test_parcel_GET(self):
```
Comment (diferpaho) [inline main/tests/test_views.py:7]: that test should be created in `harvest / tests / test_views.py`

File: sources/templates/bunch.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{% extends "base.html" %}
+
```
Comment (diferpaho) [inline sources/templates/bunch.html:2]: `bunch.html `must be created in `harvest / templates / bunch.html`

File: sources/templates/bunchbatch.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
+{% extends "base.html" %}
+
```
Comment (diferpaho) [inline sources/templates/bunchbatch.html:2]: bunchbatch.html  must be created in `harvest / templates / bunchbatch.html `

File: sources/templates/bunchcategory.html [hunk 4]
```diff
@@ -0,0 +1,86 @@
+{% extends "base.html" %}
+
```
Comment (diferpaho) [inline sources/templates/bunchcategory.html:2]: bunchcategory.html  must be created in `harvest / templates / bunchcategory.html`

File: sources/templates/sensor.html [hunk 5]
```diff
@@ -0,0 +1,82 @@
+{% extends "base.html" %}
+
```
Comment (diferpaho) [inline sources/templates/sensor.html:2]: sensor.html  must be created in `sensor.html`

File: sources/urls.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # general imports
 from django.urls import path
-from sources.views import vehicles, origin
+from sources.views import vehicles, parcel,bunchbatch, bunch,sensor,bunchcategory
```
Comment (diferpaho) [inline sources/urls.py:3]: parcel, bunchbatch, bunch, sensor, bunchcategory must be imported and used in `harvest / urls.py`

File: sources/views.py [hunk 7]
```diff
@@ -7,5 +7,17 @@ def vehicles(request):
     context = {'vehicles':vehicles}  
     return render(request, "vehicles.html", context)
```
Comment (diferpaho) [inline sources/views.py:9]: parcel, bunchbatch, bunch, sensor, bunchcategory must be imported and used in `harvest / views.py`

**Review by diferpaho — COMMENTED** [review]
review and correct everything

**Comment by osbbarr1** [comment]
I check and correct everything

File: sources/urls.py (hunk 6 above)
Comment (osbbarr1) [reply sources/urls.py]: It is added and imported and used in the harvest

File: sources/templates/bunchcategory.html (hunk 4 above)
Comment (osbbarr1) [reply sources/templates/bunchcategory.html]: I create the template bunchcategory

File: sources/templates/bunchbatch.html (hunk 3 above)
Comment (osbbarr1) [reply sources/templates/bunchbatch.html]: I create the template bunchbatch

File: sources/templates/bunch.html (hunk 2 above)
Comment (osbbarr1) [reply sources/templates/bunch.html]: I create the template bunch

File: main/tests/test_views.py (hunk 1 above)
Comment (osbbarr1) [reply main/tests/test_views.py:7]: I create the template test bunch,bunchcategory,bunchbranch etc

File: sources/templates/bunchbatch.html (hunk 3 above)
Comment (osbbarr1) [reply sources/templates/bunchbatch.html]: I create the template bunchbach

File: sources/templates/sensor.html (hunk 5 above)
Comment (osbbarr1) [reply sources/templates/sensor.html]: I create the template sensor

File: sources/views.py (hunk 7 above)
Comment (osbbarr1) [reply sources/views.py:9]: I create the template harvers / view

**Comment by osbbarr1** [comment]
Stay tuned to your comments

**Comment by diferpaho** [comment]
It gives me problems to approve a closed pull request, please create the pull request with return to the changes to be able to approve it.

**Comment by osbbarr1** [comment]
Good morning Diego, please indicate in which documentation or video I can specifically validate the request. Thanks

**Comment by osbbarr1** [comment]
I make the modification of the file in the main eliminating the testplace

**Pull request closed without merging**